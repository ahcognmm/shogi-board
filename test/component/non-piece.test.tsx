import * as React from 'react';
import {
  EmpElm,
  PromotionConfirmElm,
  Promote,
  NotPromote,
} from '../../src/components/non-piece';
import PieceObj from '../../src/game-handler/piece';
import PromotionConfirmObj from '../../src/game-handler/promotion-confirm';

describe('Emp', async () => {
  test('作れる', () => {
    const emp = <EmpElm onClick={() => console.log('onclick')} />;
    expect(emp).toBeDefined();
  });
});

describe('PromotionConfirm', async () => {
  test('作れる', async () => {
    const piece1 = new PieceObj('と', 0, 2, 7);
    const piece2 = new PieceObj('歩', 0, 2, 7);
    const pc = new PromotionConfirmObj(piece2, piece2, piece1);
    const elm = (
      <PromotionConfirmElm pcObj={pc} onClick={t => console.log('onclick')} />
    );
    expect(elm).toBeDefined();
  });
});

describe('Promote', async () => {
  test('作れる', () => {
    const elm = <Promote onClick={() => console.log('onclick')} />;
    expect(elm).toBeDefined();
  });
});

describe('NotPromote', async () => {
  test('作れる', () => {
    const elm = <NotPromote onClick={() => console.log('onclick')} />;
    expect(elm).toBeDefined();
  });
});
