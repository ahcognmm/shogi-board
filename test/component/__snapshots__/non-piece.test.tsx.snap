// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emp 作れる 1`] = `
<div
  className="piece"
  onClick={[Function]}
/>
`;

exports[`NotPromote 作れる 1`] = `
<div
  className="confirm-button"
  id="cbtn-not-promote"
  onClick={[Function]}
/>
`;

exports[`Promote 作れる 1`] = `
<div
  className="confirm-button"
  id="cbtn-promote"
  onClick={[Function]}
/>
`;

exports[`PromotionConfirm 作れる 1`] = `
<div
  className="piece"
>
  <div
    className="confirm-button"
    id="cbtn-promote"
    onClick={[Function]}
  />
  <div
    className="confirm-button"
    id="cbtn-not-promote"
    onClick={[Function]}
  />
</div>
`;
